addon/components/mobile/flexberry-objectlistview.js

```diff
@@ -138,4 +140,159 @@ export default FlexberryObjectlistview.extend({
     @default true
   */
   columnsWidthAutoresize: true,
+  /**
+    Array of objects corresponding to list of pages.
+
+    Each page is presented as object with following properties:
+    - **number** - Number of page.
+    - **isCurrent** - Page is current.
+    - **isEllipsis** - If `true` this page not showing in list.
+
+    @property _pages
+    @type Array
+    @readOnly
+  */
+
+  mobilePages: Ember.computed('pages', function() {
+    let mobilePages = [];
+    let pages = this.get('pages');
+    let currentPageNumber = pages.find(page => page.isCurrent).number;
+
+    // Rebuilding pages for mobile version.
+    switch (currentPageNumber) {
+      case 1:
+      case 2:
+        mobilePages.push(pages[0], pages[1], pages[2]);
+        break;
+      case 3:
+        mobilePages.push(pages[1], pages[2], pages[3]);
+        break;
+      case pages[pages.length - 2].number:
+      case pages[pages.length - 1].number:
+        mobilePages.push(pages[pages.length - 3], pages[pages.length - 2], pages[pages.length - 1]);
+        break;
+      default:
+        mobilePages.push(pages[2], pages[3], pages[4]);
+        break;
+    }
+
+    return mobilePages;
+  }),
+
+  /**
+    Array of pages without current.
+
+    @property _allPages
+    @type Array
+    @readOnly
+  */
+  _allPages: Ember.computed('pages', function() {
+    let allPages = [];
+    let mobilePagesNumbers = Ember.A();
+    let mobilePages = this.get('mobilePages');
+    mobilePages.forEach((page) => {
+      mobilePagesNumbers.push(page.number);
+    });
+
+    for (let i = 1; i <=  this.get('recordsTotalCount'); i++) {
+      if (!mobilePagesNumbers.contains(i)) {
+        allPages[i] = i;
+      }
+    }
+
+    return allPages;
+  }),
+
+  /*
+    Convert array of object sorting to array.
+
+    @property _currecntSortingArray
+    @type Array
+    @readOnly
+  */
+  _currecntSortingArray: Ember.computed('sorting',  function() {
+    let sorting = this.get('sorting');
+    let sortingKeys = Object.keys(this.get('sorting'));
+    let columns = Ember.A();
+    sortingKeys.forEach(key => {
+      let column = sorting[key];
+      columns.pushObject({
+        key: key,
+        sortNumber: column.sortNumber,
+        sortAscending: column.sortAscending
+      });
+    });
+    columns = columns.sort((a, b) => a.sortNumber - b.sortNumber);
```

Comment:
Сортировка выполняется "на месте", т.е. изменяет сам массив.
Если используется специальный массив, то почему бы не воспользоваться его возможностями.
```suggestion

    columns.sortBy('sortNumber');
```